azure-quantum/azure/quantum/optimization/solvers.py

```diff
@@ -78,6 +78,46 @@ def __init__(
     SWEEPS_WARNING = 10000
     TIMEOUT_WARNING = 600
 
+    def _jobdetails_from_problem(
```

Comment:
This is a nice refactor; however, we currently have a feature branch open where we've refactored this in a different way (see #108), I suggest let's wait until that is merged into `main` and pull in the changes here

Reply:
It is now refactored, see here for the `Solver.submit` method:
https://github.com/microsoft/qdk-python/blob/1f2607e0c22a27417635a98991fbd9f803dc34d2/azure-quantum/azure/quantum/optimization/solvers.py#L76